### DISCUSSION TITLE
Nohup in Unix

### DISCUSSION THREAD

**Discussion by joindevopscloud** [daws-76s/.github#4] [Q&A]
What is the purpose of 'nohup' in Unix?

**Comment by saipdevops555** [comment]
nohup is a command in Unix and Unix-like operating systems that stands for "no hang up." Its primary purpose is to execute a command and ensure that it continues running even if the user who started it logs out or the terminal is closed.

Using nohup is particularly useful for running long-running processes or background tasks that you want to persist even after you've logged out or closed your terminal session.

Here's a basic usage example:
**nohup command-to-run &

command-to-run: The command or script you want to execute.
&: Puts the command in the background.**

The nohup command redirects the standard output and standard error to a file named nohup.out by default, so any output from the command won't interfere with the terminal.

**Comment by balajibigdatadevops** [comment]
nohup command is run the script or command runs background.
by default any command runs in foreground. nohup means even if terminate closed or abruptly terminal ur session.
nohup being used, its stands for no hang up.

ex: nohup ./your_script.sh > script_output.log 2>&1 &
once we we run above command, it generates process ID. that we can verify using ps -ef | grep <PID> or above log also to monitor the status.

Best use case:
long running scripts. Production environment we used mostly nohup to kick off command or scripts in background.
Also if script or running background, we can work on other tasks as usual, no need to wait until that is completed.
